File: Java/Edwin Villarraga/TernarySearch.java
public class Main {
	static int ternarySearch(int left,int r,int key,int arr[]){
		if(r>=left){
			int mid1 = left + (r-left)/3;
			int mid2 = r - (r-left)/3;
			if (arr[mid1]==key){
				return mid1;
			}
			if(arr[mid2]==key){
				return mid2;
			}if(key<arr[mid1]){
				return ternarySearch(left,mid1-1,key,arr);
			}else if(key>arr[mid2]){
				return ternarySearch(mid2+1,r,key,arr);
			}else{
				return ternarySearch(mid1+1,mid2-1,key,arr);
			}
			
		}
		return -1;
	}
	public static void main(String[] args) {
		int arr[]={1,2,3,4,5,6,7,8,9};
		int l=0;
		int r=9;
		int x=5;
		int result = ternarySearch(l,r,x,arr);
		if(result<0){
			System.out.println("el numero "+x+" no se encuentra en el array");
		}else{
		System.out.println("indice encontrado en la posicion "+result);
	}
	}
}

File: PY/Bryann Valderrama/ternarySearch.py
def ternarySearch(izq, der, num, lista):
    if (der >= izq):
        mid1 = int(izq+(der-izq)/3)
        mid2 = int(der-(der-izq)/3)
        if (lista[mid1] == num):
            return mid1
        if (lista[mid2] == num):
            return mid2
        if (num < lista[mid1]):
            return ternarySearch(izq, mid1-1, num, lista)
        elif (num > lista[mid2]):
            return ternarySearch(mid2+1, der, num, lista)
        else:
            return ternarySearch(mid1+1, mid2-1, num, lista)
    return -1


if __name__ == '__main__':
    lista = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
    izq = int(0)
    der = len(lista)-1
    numeroabuscar = int(input())
    resultado = ternarySearch(izq, der, numeroabuscar, lista)
    if(resultado == -1):
        print('El numero no esta en la lista')
    else:
        print(f'Encontrado en el indice {resultado}')
